release/src/module_dependencies/main.py

```diff
@@ -0,0 +1,172 @@
+            G.add_edge(module['name'], dependent)
+
+    processingList = []
+
+    # Nodes with no in degrees = 0 and out degrees != 0 are marked as level 1 and the node is appended to the processing list
+    for root in [node for node in G if G.in_degree(node) == 0 and G.out_degree(node) != 0]:
+        G.nodes[root]['level'] = 1
+        processingList.append(root)
+
+    # While the processing list is not empty, successors of each node in the current level are determined
+    # For each successor of the node, 
+    # 			longest path from node to successor is considered and intermediate nodes are removed from dependent list - removeModulesInIntermediatePaths
+    # 			the level is updated and the successor is appended to a temporary array
+    # After all nodes are processed in the current level the processing list is updated with the temporary array
+    level = 2
+    while len(processingList) > 0:
+        temp = []
+        for node in processingList:
+            successors = []
+            for i in G.successors(node):
+                successors.append(i)
+
+            for successor in successors:
+                
+                removeModulesInIntermediatePaths(G, node, successor, successors, moduleDetailsJSON)
+
+                G.nodes[successor]['level'] = level
+                if successor not in temp:
+                    temp.append(successor)
+        processingList = temp
+        level = level + 1
+
+    for module in moduleDetailsJSON['modules']:
+        module['level'] = G.nodes[module['name']]['level']
+
+    return moduleDetailsJSON
+
+
+# Updates the stdlib_modules.JSON file with dependents of each standard library module
+def updateJSONFile(updatedJSON):
+    with open('./release/resources/stdlib_modules.json', 'w') as jsonFile:
+        jsonFile.seek(0) 
+        json.dump(updatedJSON, jsonFile, indent=4)
+        jsonFile.truncate()
+
+# Creates a JSON string to store module information
+# returns: JSON with module details
+def initializeModuleDetails(moduleNameList):
+
+    moduleDetailsJSON = {'modules':[]}
+
+    for moduleName in moduleNameList:
+        version = getVersion(moduleName)						
+        moduleDetailsJSON['modules'].append({
+            'name': moduleName, 
+            'version':version,
+            'level': 0,
+            'release': True, 
+            'dependents': [] })
+
+    return moduleDetailsJSON
+
+# Gets all the dependents of each module to generate the dependency graph
+# returns: module details JSON with updated dependent details
+def getImmediateDependents(moduleNameList, moduleDetailsJSON):
+    for moduleName in moduleNameList:
+    
+        dependencies = getDependencies(moduleName)
+
+        for module in moduleDetailsJSON['modules']:
+            if module['name'] in dependencies:
+                moduleDetailsJSON['modules'][moduleDetailsJSON['modules'].index(module)]['dependents'].append(moduleName)
+                    
+    return moduleDetailsJSON
+
+def main():
```

Comment:
Shall we move this `main` function to the top?

Reply:
Updated